src/RestApi/StoreApi/Controllers/ProductCollectionData.php

```diff
@@ -143,27 +143,71 @@ public function get_items( $request ) {
 		$filters = new ProductQueryFilters();
 
 		if ( ! empty( $request['calculate_price_range'] ) ) {
-			$price_results = $filters->get_filtered_price( $request );
+			$filter_request = clone $request;
```

Comment:
This is just a shallow clone - is that known and desired here?  It's not immediately clear to me why this is needed?  The concern I have is that I see `$filter_request` is cloned in multiple condition blocks that are not mutually exclusive (as far as I can tell) so is it intentional that subsequent clones wipe out previous?

Reply:
I wanted to avoid touching the original query, which was happening before I added the clone.

Reply:
So would there be value in cloning once then? As it is now, it appears in at least one spot you could be cloning a clone.

Reply:
We are cloning `$request` into a new variable called `$filter_request` and then using the `set_param` method. We don't want to change the original `$request` object. Actions pass references, not cloned objects.

Reply:
Gotcha, the line I was thinking of was was 172 and I completely glossed over that it was cloning the `$request` again.  For some reason I thought it was doing `$filter_request = clone $filter_request` 🤦‍♂